### ISSUE TITLE
508-defect-1❗️[SCREEN READER]: Lang attribute must be accurate within the "in this section" subnavigation

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22518]
# [508-defect-1 :exclamation:](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _First name only. Delete if not applicable._

## User Story or Problem Statement

As a Spanish or Filipino screen reader user, I want to be able to understand the "in this section" subnavigation.

## Details

To meet Section 508 compliance, the lang attribute must be accurate to the language on the page/parts. 

Currently, the in this section subnavigation is nested within a `div` that carries the language attribute of the page.

```html
<div id="content" lang="es"> <!-- All children content will be pronounced in a Spanish example -->
  <nav>
    <!-- Subnav content here -->
  </nav>
  <!-- Actual page content follows here-->
</div>
```

Since the majority of the subnav's content is in English, Spanish or Filipino screen reader users will hear an incomprehensible dialog of English pronounced in the accent denoted in the parent `div`. 

This may result in abandonment of the page before the actual content is reached.

https://user-images.githubusercontent.com/14154792/113320162-32f24580-92e0-11eb-9793-a66f2799a7c5.mp4

## Acceptance Criteria

- [ ] All elements within "In this section" subnavigation are pronounced with the proper accent

## Steps to Recreate

1. Enter `<https://staging.va.gov/health-care/covid-19-vaccine-esp/>` in browser
2. Start screenreading device listed in Environment
3. Tab to the subnavigation
4. Confirm English is read out in an Spanish accent 

### Short term solution

This subnav is currently wrapped within a div that sets the language of the page. Since the majority of the subnav is in English, I would recommend to add the attribute `lang="en"` to the `nav` element and then add `lang` attributes where needed for children elements _not_ in English. For example, if a user is on the Spanish version of the COVID-19 page:

```html
<!-- All children elements are pronounced in English unless explicitly set to a different language -->
<nav lang="en"> 
  <ul>
    <li><a>COVID-19 vaccines at VA</a></li>
    <!-- Since this is in Spanish, we explicitly set the lang attribute to Spanish -->
    <li><a lang="es">Vacunas contra COVID-19 en el VA</a></li> 
    <!-- Since this is in Tagalog, we explicitly set the lang attribute to Tagalog -->
    <li><a lang="tl">Mga bakuna sa COVID-19 sa VA</a></li>
    </ul>
</nav>
```

### Long term solution

If this page is truly meant to help our Spanish / Tagalog speaking Veterans/family members, the _entire_ page should be translated including global components. The `html` should carry the `lang` attribute instead of a content div.

This will also make this page more findable to Spanish and Tagalog speaking users as it improves SEO.

## WCAG or Vendor Guidance (optional)

* [3.1.1 – Language of Page (Level A)](https://www.wuhcag.com/language-of-page/)
* [3.1.2 – Language of Parts (Level AA)](https://www.wuhcag.com/language-of-parts/)

**Comment by mnorthuis** [comment]
This seems like a very logical interim solution for this.  The longer term solution to translate the entire page/site is currently in early discovery.

**Comment by SKasimow** [comment]
@mnorthuis thank you Mikki.

@youngfreezyVA - can you review as well?

CC @DanielleThierryUSDSVA

**Comment by youngfreezyVA** [comment]
this looks like a good short term solution, since we have the links `href`, which follows established patterns for languages (ends with -esp or -tag). if this element were not a link, we would need an interim (CMS adds the lang attribute to this field) solution.

**Comment by SKasimow** [comment]
cc: @DanielleThierryUSDSVA FYI

**Comment by SKasimow** [comment]
@joshkimux Hi - can we close this one?

**Comment by joshkimux** [comment]
@SKasimow , yes this looks to be set!
